### ISSUE TITLE
Add support for browser-sync 3

### ISSUE THREAD

**Issue by jkevingutierrez** [Va1/browser-sync-webpack-plugin#97]
browser-sync-webpack-plugin requires browsery-sync^2 as peer-dependency https://github.com/Va1/browser-sync-webpack-plugin/blob/master/package.json#L23 

Now that browser-sync 3 was released, it generates a dependency conflict when both packages are used

<img width="1081" alt="image" src="https://github.com/Va1/browser-sync-webpack-plugin/assets/6220055/692044a6-4bf6-4a39-9bc0-f6862f88da97">

**Comment by nlenkowski** [comment]
Just ran into this dependency conflict as well, when upgrading a project to browsersync@3.

**Comment by elrosi** [comment]
Hi, I have the same problem.

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: browser-sync-webpack-plugin@2.3.0
npm ERR! Found: browser-sync@3.0.2
npm ERR! node_modules/browser-sync
npm ERR!   dev browser-sync@"^3.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer browser-sync@"^2" from browser-sync-webpack-plugin@2.3.0
npm ERR! node_modules/browser-sync-webpack-plugin
npm ERR!   dev browser-sync-webpack-plugin@"^2.3.0" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: browser-sync@2.29.3
npm ERR! node_modules/browser-sync
npm ERR!   peer browser-sync@"^2" from browser-sync-webpack-plugin@2.3.0
npm ERR!   node_modules/browser-sync-webpack-plugin
npm ERR!     dev browser-sync-webpack-plugin@"^2.3.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

**Comment by goxmedia** [comment]
Same here:

warning " > browser-sync-webpack-plugin@2.3.0" has incorrect peer dependency "browser-sync@^2".

**Comment by Jas-n** [comment]
There's an updated repo on npm via npm install --save-dev browser-sync-v3-webpack-plugin

**Comment by AdamReznicek** [comment]
Same here

**Comment by caleb-pierce** [comment]
@Jas-n This did not work for me. As soon as I attempted to mix:
```
watch    
> mix watch

        Additional dependencies must be installed. This will only take a moment.
                                                                                             
        Running: npm install browser-sync-webpack-plugin@^2.3.0 --save-dev --legacy-peer-deps
                                                                                             
        Finished. Please run Mix again.

```

**Comment by Jas-n** [comment]
> @Jas-n This did not work for me. As soon as I attempted to mix:
> ```
> watch    
> > mix watch
> 
>         Additional dependencies must be installed. This will only take a moment.
>                                                                                              
>         Running: npm install browser-sync-webpack-plugin@^2.3.0 --save-dev --legacy-peer-deps
>                                                                                              
>         Finished. Please run Mix again.
> 
> ```

That's normal for the first run.

Just run the run command again. You won't see it next time 😊.

**Comment by brvm** [comment]
@Jas-n  when running `npm install --save-dev browser-sync-v3-webpack-plugin` and then calling `.browserSync()`, it will still say:

```
watch    
> mix watch

        Additional dependencies must be installed. This will only take a moment.
                                                                                             
        Running: npm install browser-sync-webpack-plugin@^2.3.0 --save-dev --legacy-peer-deps
                                                                                             
        Finished. Please run Mix again.
```

Which then proceeds to install the older version and the issue still persists.

**Comment by FeBe95** [comment]
I can confirm that the fix mentioned by @Jas-n does not work when using it together with the `laravel-mix` package. Manual installation of the new package works fine, but `laravel-mix` always installs the old `browser-sync-webpack-plugin` automatically again. Unfortunately I didn't find a way to prevent this.

**Comment by caleb-pierce** [comment]
> I can confirm that the fix mentioned by @Jas-n does not work when using it together with the `laravel-mix` package. Manual installation of the new package works fine, but `laravel-mix` always installs the old `browser-sync-webpack-plugin` automatically again. Unfortunately I didn't find a way to prevent this.

I am having the same issue. Security vulnerability still persists.

**Comment by TJCrouch** [comment]
> > I can confirm that the fix mentioned by @Jas-n does not work when using it together with the `laravel-mix` package. Manual installation of the new package works fine, but `laravel-mix` always installs the old `browser-sync-webpack-plugin` automatically again. Unfortunately I didn't find a way to prevent this.
> 
> I am having the same issue. Security vulnerability still persists.
> 
> Could we get some other more people to review [mozilla/ssl-config-generator#235](https://github.com/mozilla/ssl-config-generator/pull/235) and abandon this project switching to a fork that supports v3? I am unsure of the etiquette here as this project hasn't had a release since 2020.

Added my review to it.  Hopefully this will pick up more traction!

**Comment by joshlewis** [comment]
@Va1 Is it fair to ask for a new tag to be made so everyone gets this fix? 2.3.1 or something?
